src/features/exporter/Downloader.js

```diff
@@ -25,18 +26,21 @@ const mapStateToProps = (state, ownProps) => {
     shape: state.shapes.currentId,
     offsetX: (state.machine.minX + state.machine.maxX) / 2.0,
     offsetY: (state.machine.minY + state.machine.maxY) / 2.0,
+    width:   (state.machine.rectangular ? (state.machine.maxX - state.machine.minX) : (2.0 * state.machine.maxRadius)),
```

Comment:
I've been a bit lazy, because most users use rectangular limits with gcode and polar limits with thr, but there's no specific requirement for that. Either it needs to be explicitly coupled somehow, or support criss-cross of these.